src/iGenVar.cpp

```diff
@@ -13,116 +13,163 @@
+
+static inline std::vector<std::string> combined_extensions{
+    []()
+    {
+        if (compression_extensions.empty())
+            return sequence_extensions; // GCOVR_EXCL_LINE
+        std::vector<std::string> result;
+        for (auto && sequence_extension : sequence_extensions)
+        {
+            result.push_back(sequence_extension);
+            for (auto && compression_extension : compression_extensions)
+                result.push_back(sequence_extension + std::string{'.'} + compression_extension);
+        }
+        return result;
+    }()};
+
+void initialize_argument_parser(sharg::parser & parser, cmd_arguments & args)
 {
     parser.info.author = "Lydia Buntrock, David Heller, Joshua Kim";
     parser.info.app_name = "iGenVar";
     parser.info.man_page_title = "Short and Long Read SV Caller";
     parser.info.short_description = "Detect genomic variants in a read alignment file";
     parser.info.version = "0.0.3";
-    parser.info.date = "30-03-2021";    // last update
+    parser.info.date = "11-07-2022";    // last update
     parser.info.email = "lydia.buntrock@fu-berlin.de";
     parser.info.long_copyright = "long_copyright";
     parser.info.short_copyright = "short_copyright";
     parser.info.url = "https://github.com/seqan/iGenVar/";
 
     // Options - Input / Output:
-    parser.add_option(args.alignment_short_reads_file_path,
-                      'i', "input_short_reads",
-                      "Input short read alignments in SAM or BAM format (Illumina).",
-                      seqan3::option_spec::standard,
-                      seqan3::input_file_validator{{"sam", "bam"}} );
-    parser.add_option(args.alignment_long_reads_file_path,
-                      'j', "input_long_reads",
-                      "Input long read alignments in SAM or BAM format (PacBio, Oxford Nanopore, ...).",
-                      seqan3::option_spec::standard,
-                      seqan3::input_file_validator{{"sam", "bam"}} );
-    parser.add_option(args.genome_file_path,
-                      'g', "input_genome",
-                      "Input the sequence of the reference genome.",
-                      seqan3::option_spec::standard,
-                      seqan3::input_file_validator<seqan3::sequence_file_input<>>{} );
-    parser.add_option(args.output_file_path, 'o', "output",
-                      "The path of the vcf output file. If no path is given, will output to standard output.",
-                      seqan3::option_spec::standard,
-                      seqan3::output_file_validator{seqan3::output_file_open_options::open_or_create, {"vcf"}});
+    parser.add_option(args.alignment_short_reads_file_path, sharg::config{
+                        .short_id = 'i', .long_id = "input_short_reads",
+                        .description = "Input short read alignments in SAM or BAM format (Illumina).",
+                        .advanced = false,
```

Comment:
```suggestion
```
you don't need this as `false` is the default. Or do you want to be explicit here?